tmk_core/avr.mk

```
@@ -160,6 +160,8 @@ define EXEC_AVRDUDE
 	list_devices() { \
 		if $(GREP) -q -s icrosoft /proc/version; then \
 		    wmic.exe path Win32_SerialPort get DeviceID 2>/dev/null | LANG=C perl -pne 's/COM(\d+)/COM.($$1-1)/e' | sed 's!COM!/dev/ttyS!' | xargs echo -n | sort; \
+		elif [ "`uname`" = "FreeBSD" ]; then \
+			ls /dev/tty* | grep -v '\.lock$$' | grep -v '\.init$$'; \
```

This doesn't actually work for me.
But this one does:
```suggestion
			ls /dev/tty* | grep -vE '\.(lock|init)$'; \
```

---

Are you sure that there should be only one `$` there? `make` is going to expand it to the value of variable `$'` instead of a literal `$`.

---

BTW, I've gone for two grep's for simplicity and clarity. If we want, we can replace everything here with a `find` call instead:

```
find /dev/tty* -not -name '*lock' -not -name '*init'
```

---

Yeah, you are right, `make` will expand it.
I could not test the PR in it's entirety today, just run the line on a FreeBSD VM. I'll check it out tomorrow in more detail.